### PR TITLE
feat(python,rust): missing column width can be set exactly (not as a lower bound)

### PR DESCRIPTION
**Pull request by alexander-beedie**
While investigating column width changes in the latest comfy-table (ref: https://github.com/pola-rs/polars/pull/9656) I saw that we can set the "missing column" width exactly as it is always a fixed/exact width; this will ensure that it doesn't get allocated any extraneous potential width from the table's total width budget.

This is a very minor update; a more comprehensive update of our comfy-table constraints will follow later, which should improve width determination and table repr in the general case, reducing instances of unnecessary/premature column/text wrapping. Currently we only set _lower_ bound constraints, but there are places we should also set _upper_ bound constraints (such as date columns, since we know that a date cannot exceed `yyyy-mm-dd` chars), or even absolute constraints in a few cases.

(Actual impact - where we omit some columns, an extra character or two will be available for assignment into the _other_ columns' potential width buckets).

### CODE REVIEW

File: polars/polars-core/src/fmt.rs [hunk 1]
```diff
@@ -461,24 +461,29 @@ impl Display for DataFrame {
                 };
                 (s, lower_bounds)
             };
-            let tbl_lower_bounds =
+
+            let col_width_exact =
+                |l: usize| ColumnConstraint::Absolute(comfy_table::Width::Fixed(l as u16));
+            let col_width_lower_bound =
                 |l: usize| ColumnConstraint::LowerBoundary(comfy_table::Width::Fixed(l as u16));
+            // let col_width_upper_bound =
```
Comment (ritchie46) [inline polars/polars-core/src/fmt.rs:469]: Commented out code. Don't we use this anymore?

Comment (alexander-beedie) [reply polars/polars-core/src/fmt.rs:469]: Ahh, minor oversight there - it’s new; I’m going to use it in a follow-up patch (we’ve not previously set upper bounds).

Comment (stinodego) [reply polars/polars-core/src/fmt.rs:469]: Let's get rid of the commented out code for now, keep the main branch clean. Then I'll approve this!

Comment (alexander-beedie) [reply polars/polars-core/src/fmt.rs]: Done; had to get off the side of a Swiss mountain first :))

**Review by stinodego — APPROVED** [review]

**Pull request merged**